### ISSUE TITLE
Implement `jsonSerializable` on the Layout collection

### ISSUE THREAD

**Issue by harmenjanssen** [whitecube/nova-flexible-content#256]
Context: we're using Laravel Nova as a headless CMS and disclose information through a JSON API. The rendering happens in a different app (next.js in this case, but could be anything).

When rendering the JSON for the API, the `LayoutCollection` will be transformed into an empty array.   
I can call `$model->content->toArray()`, and this will give me the content, but it loses the names of the layouts:

```json
[
  {
    "title": "Foo",
    "content": "Lorem ipsum"
  },
  {
    "title": "Bar",
    "content": "Lorem ipsum"
  }
]
```

This way our front-end doesn't know which layout (template) to render.

I was wondering whether you're open to accepting a Pull Request which implements `JsonSerializable` on the collection, which transforms the output to something like this:

```
[
  {
    "type": "myLayout",
    "attributes": {
      "title": "Foo",
      "content": "Lorem ipsum"
    }
  },
  {
    "type": "myLayout",
    "attributes": {
      "title": "Bar",
      "content": "Lorem ipsum"
    }
  }
]
```

If so, I will happily submit a PR.

**Comment by chrishilmes** [comment]
Yeah i'm in exactly the same situation and would love this feature out of the box. Just one thing: `type` is a reserved keyword in JSON API. So `layout` could be the better keyword for the name of the layout.

**Comment by harmenjanssen** [comment]
@chrishilmes That's good to know, thanks for sharing. The final format of the JSON would be something to discuss, of course.

**Comment by Nyratas** [comment]
Hi,

I like the idea, but would prefer not to have different data structures or keywords in different situations in order to keep things predictable (if we use "type" in the database & PHP layouts collection, it should be "type" in the JSON too). Since that seems to be a problem, we could change "type" into "layout" everywhere, but that would be a breaking change.

Such changes can be done but would mean a new major release of the package. I'd suggest to merge your PR in a new  `1.x` branch that we would keep up-to-date with other planned breaking changes (and proper testing 🤞) until release.

So yeah, I would be happy to review your PR! Thanks for the suggestion.

**Comment by harmenjanssen** [comment]
Alright, good to hear. I understand your concern about the consistency of the keywords.

I'm going to flesh out a version of this locally in a project, and then submit a PR when I think it makes sense and includes everything one would need in my situation.

One other aspect of this is actually how media is stored currently. At the moment I can't look at the JSON and know which image to render. 
For this feature to be really useful the `jsonSerialize` method should also do something smart with media, in a way the front-end can render the images.

I will get back to you on this!

**Comment by harmenjanssen** [comment]
I tried to spend some time on this last night, and then noticed `Layout` already has a `jsonSerialize` method! It would make more sense to use that than to make my own new thing. 
However, its output doesn't look like my proposed format above, so I will try and figure out if I can change it safely.

Yesterday I ran into some trouble with nested flexible layouts. It's possible to cast nested layouts by implementing accessor methods on the Layout classes, though. All in all it's quite a balancing act between different parts of the system that all have to work together otherwise you'd still get a `null` or empty array somewhere in the JSON output. 

I'm working on a client project right now that uses this package extensively, nested, unnested, with and without media, et cetera, so hopefully after this project I have a good idea of what's necessary to fix this.
